h/streamer/messages.py

```diff
@@ -140,8 +147,8 @@ def _generate_annotation_event(message, socket, annotation, user_nipsad):
     base_url = socket.registry.settings.get('h.app_url',
                                             'http://localhost:5000')
     links_service = LinksService(base_url, socket.registry)
-    serialized = presenters.AnnotationJSONPresenter(annotation,
-                                                    links_service).asdict()
+    resource = AnnotationResource(annotation, group_service)
```

Comment:
Caching the group object within the handling of a single message is fine, but caching between messages is bound to end in disaster -- if someone changes the sharing settings of a group that's cached we'll have to restart the websocket server before it starts calculating permissions correctly again.